code/modules/zombie/organs.dm

```diff
@@ -0,0 +1,87 @@
+#define START_TIMER reanimation_timer = world.time + rand(600,1200)
+
+/obj/item/organ/internal/zombie_infection
+	name = "festering ooze"
+	desc = "A black web of pus and vicera."
+	parent_organ = "head"
+	slot = "zombie_infection"
+	icon_state = "blacktumor"
+	var/causes_damage = TRUE
+	var/reanimation_timer
+	var/datum/species/old_species = /datum/species/human
```

Comment:
You need to set the mob's actual species to this var in `New()` - otherwise you'll turn into a human after the organ is removed, regardless of your initial species

Reply:
It sets it before their species is changed in the zombify proc